### PR TITLE
Client Del fix.

### PR DESCRIPTION
**Pull request by AnturK**
Weird side-effects here we go.

Moved the removal to the top because that's pretty much most important thing here.

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -453,6 +453,10 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 //////////////
 
 /client/Del()
+	GLOB.clients -= src
+	GLOB.directory -= ckey
+	if(!gc_destroyed)
+		Destroy() //Clean up signals and timers.
```
Comment (Cyberboss) [inline code/modules/client/client_procs.dm]: :thonk:

If we're going to call Destroy from Del should the bulk of the code not be moved there?

Comment (AnturK) [reply code/modules/client/client_procs.dm]: Probably, yeah. I'll do it in a moment.

Comment (optimumtact) [inline code/modules/client/client_procs.dm]: this shouldn't even be possible, what is calling del these days?

**Comment by Cyberboss** [comment]
It's called when the client dcs

**Review by SpaceManiac — APPROVED** [review]

**Comment by Cyberboss** [comment]
**screaming internally**

**Comment by AnturK** [comment]
?

**Pull request merged**